src/components/CreateTask/CreateTask.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from "react";
+import "./CreateTask.css";
+import { useNavigate, Link } from "react-router-dom";
+import { createTask } from "../../utils/api";
+
+const CreateTask = () => {
+  const navigate = useNavigate();
+  const [task, setTask] = useState({
+    title: "",
+    description: "",
+    dueDate: "",
+    status: "in-progress",
+  });
+
+  const handleChange = (e) => {
+    setTask({ ...task, [e.target.name]: e.target.value });
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    try {
+      await createTask(task);
+
+      navigate("/tasks");
+    } catch (error) {
+      console.error("Error creating task:", error);
+    }
+  };
+
+  return (
+    <div className="container mt-3 create-task-container">
+      <h2>Create Task</h2>
+      <form onSubmit={handleSubmit} className="task-form">
```

Comment:
* You need to use `ModalWithForm` for any modal with a form in the project. It's the main reason why you have this unversal component